core/src/turbulence.rs

```diff
@@ -0,0 +1,240 @@
+/// This file is a Rust port of the C reference implementation of the
+/// feTurbulence element in the SVG specification. It's the usual Perlin noise.
+/// See: https://www.w3.org/TR/SVG11/filters.html#feTurbulenceElement
+/// The `octave_offsets` parameter of `turbulence` was added after porting.
```

Comment:
Per the [W3C Document License](https://www.w3.org/Consortium/Legal/2015/doc-license), let's add this notice:
`Copyright © 2015 W3C® (MIT, ERCIM, Keio, Beihang). This software or document includes material copied from or derived from https://www.w3.org/TR/SVG11/filters.html#feTurbulenceElement.`